### ISSUE TITLE
Training instability

### ISSUE THREAD

**Issue by bdevnani3** [End2End-Diffusion/iREPA#4]
Hi,

Thanks to the authors for releasing the code and for the great paper! 🙂

Similar to [lezhang7](https://github.com/End2End-Diffusion/iREPA/issues/2), I’m also seeing unstable training. Across two runs, I observe the same pattern: gradient norm spikes/explosions followed by NaNs.

<img width="1315" height="637" alt="Image" src="https://github.com/user-attachments/assets/d80f490d-00da-466a-b5bd-7009e6355702" />

**REPA**:

```
accelerate launch train.py --config configs/repa.yaml \
  --model="SiT-B/2" \
  --enc-type="pe-vit-b" \
  --encoder-depth=4 \
  --data-dir=../data \
  --exp-name="sitb2-pe-vit-b-repa" \
  --batch-size=32
```

**iREPA**:

```
accelerate launch train.py --config configs/irepa.yaml \
  --model="SiT-B/2" \
  --enc-type="pe-vit-b" \
  --encoder-depth=4 \
  --data-dir=../data \
  --exp-name="sitb2-pe-vit-b-irepa" \
  --batch-size=32
```

**Hardware**: 8× NVIDIA A40, resulting in an effective batchsize of 256.

If there are recommended stabilization tricks (e.g., gradient clipping, LR tweaks/warmup, AMP settings, etc.) or a known gotcha with this config, I’d really appreciate any guidance.

**Comment by 1jsingh** [comment]
Thanks for the question. To clarify, even baseline repa is unstable? Also batchsize is the global batchsize, so in above the effective bz is still 32 not 256. Can you please try w/ standard 256bz to debug further? We have tried on a100, h100, h200 and it seems stable.

**Comment by bdevnani3** [comment]
Thank you for the quick response! :) 

Yes indeed, the orange plot above is the baseline.

I've attempted to retrain this configuration (baseline) with your recommendation.

REPA(baseline):

```
accelerate launch train.py --config configs/repa.yaml \
  --model="SiT-B/2" \
  --enc-type="dinov2-vit-b" \
  --encoder-depth=4 \
  --data-dir=../data \
  --exp-name="sitb2-dinov2-vit-b-repa-100k-4gpus" \
  --max-train-steps=100000 \
  --batch-size=256
```

**Hardware**: 4× NVIDIA A40

**Grad Norm**:

<img width="1223" height="660" alt="Image" src="https://github.com/user-attachments/assets/0c83c1f8-01df-4843-8818-8fdc4af02522" />

**Denoising Loss**:

<img width="1194" height="663" alt="Image" src="https://github.com/user-attachments/assets/8e1e9d68-a0ab-4f52-b0a0-795225f62491" />

Any other suggestions? :)

**Comment by bdevnani3** [comment]
Following up here. It seems like it was a precision issue after all. Changing to bf16 has stabilized training. I'll add safeguards for overflow if I switch to fp16. Thanks!

<img width="1217" height="625" alt="Image" src="https://github.com/user-attachments/assets/532f2c49-f6ef-4d14-86e2-e486f962acd6" />
